Simplenote/src/main/java/com/automattic/simplenote/TagDialogFragment.java

```diff
@@ -87,12 +79,62 @@ public void onClick(DialogInterface dialog, int whichButton) {
         builder.setView(view);
         mDialogEditTag = builder.create();
         mDialogEditTag.setOnShowListener(this);
+
+        mClickListenerNegativeRename = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                dismiss();
+            }
+        };
+
+        mClickListenerPositiveRename = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String tagNew = getTagNew();
+
+                if (tagNew.equals(mTagOld)) {
+                    dismiss();
+                }
+
+                int index = mTag.hasIndex() ? mTag.getIndex() : mBucketTag.count();
+                boolean isRenamingToLexicalTag = TagUtils.hashTag(tagNew).equals(TagUtils.hashTag(mTagOld));
+                boolean hasCanonicalTag = TagUtils.hasCanonicalOfLexical(mBucketTag, tagNew);
+
+                if (hasCanonicalTag && !isRenamingToLexicalTag) {
+                    String tagCanonical = TagUtils.getCanonicalFromLexical(mBucketTag, tagNew);
+                    showDialogErrorConflict(tagCanonical, mTagOld);
+                    return;
+                }
+
+                tryToRenameTag(tagNew, index);
+            }
+        };
+
+        mClickListenerNegativeConflict = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showDialogRenameTag();
+            }
+        };
+
+        mClickListenerPositiveConflict = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String tagNew = getTagNew();
+                int index = mTag.hasIndex() ? mTag.getIndex() : mBucketTag.count();
+                tryToRenameTag(tagNew, index);
+                dismiss();
```

Comment:
Question (❓ ): Why adding the extra `dismiss()` here? This can be probably removed since `dismiss()` would anyway be triggered as part of the finally clause within `tryToRenameTag(...)`, isn't that right?

Reply:
I forgot to delete that `dismiss();` statement after adding it to the `finally` clause in `tryToRename`.  I removed the extra statement in 6de2d25a.